sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1094,6 +1142,29 @@ std::vector<node> get_root_nodes() const;
 ----
 |Returns a list of all nodes in the graph which have no dependencies.
 
+|
+[source,c++]
+----
+event get_event(const node& node);
+----
+|Returns a SYCL event which represents the completion of node `node` which is
+valid only for the next execution of the graph. This event can be used as a
```

Comment:
I'm curious why the event is valid only for the `next` graph execution.  In graph.finalize(), the event is tied with a native event, and I think (at lease for most cases), the event is fixed after graph.finalize(). 

Or the actual purpose of this function is to get a chance to reset the fixed event?

Reply:
> It is not necessarily true that the backend event will be reset by this function call, but that is the general idea.

Does this mean that "the backend event will be reset by this function call" is the general idea in practice, but it is not a must?  (From spec perspective, it does not define this function's implementation.)


Suppose one implementation chooses to reset the backend event for this function, and the event is not reused by other places, based on it, just a "thought experiment",  the code below:
```
dynEvent.update(execGraphA.get_event(nodeA));
execGraphB.update(nodeB);
```
can be simplified with below code:
```
execGraphA.get_event(nodeA);
```
Is it correct for this special implementation? The reason is that the backend event used by execGraphA does not change (just the event status is changed), and so the two "update" functions can be removed for a certain implementation.

Reply:
I think my previous reply was poorly worded, I did not mean to imply that that is how the implementation will behave, and I don't think an implementation would or should allow your simplified code.

This is basically the same idea being discussed [here](https://github.com/reble/llvm/pull/372#discussion_r1617576585) if I'm not mistaken, and I would reiterate that is something we can explore in the future but not as part of this initial work.

Reply:
what's the expected/typical implementation of this function to make the returned event only valid for the next execution of the graph, thanks.

Reply:
I'm afraid I don't really have an answer for that immediately, but I think you are right that this needs some further consideration so we are looking into it.

Reply:
Considering this more, I'm concerned that `get_event(node)` may not be implementable on a CUDA backend. Since the executable graph has already been finalized then we can't add nodes to it, but to implement this ontop of CUDA-Graph I think we need to use `cudaGraphAddEventRecordNode()` - which would require some mechanism in the modifiable graph to say that a node is going to produce a sycl event.

Reply:
Wondering if defining the semantics of dynamic_event like a counter-based event would help give a more optimized flow for the node-to-node only use case. 
* If a `dynamic_event` is created from a vanilla sycl queue submission. Counter only incremented once when sycl event backing it is complete.
* If a `dynamic_event` is created for a node in a graph, then counter incremented on completion of each execution of the graph.

* When waiting on a `dynamic_event` in a vanilla queue submission, waiting on value to increment from current value.

A user wouldn't need to manually call any reset methods or the update() calls currently needed for the node-to-node between graph case. Backend implementation could use an event reset, or even native counter based events where applicable.